packages/api/src/derives/nft/openCollectionListings.ts

```diff
@@ -0,0 +1,65 @@
+// Copyright 2019-2021 Centrality Investments Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { Observable, combineLatest } from 'rxjs';
+import { map, switchMap, reduce, mergeAll, first } from 'rxjs/operators';
+import { ApiInterfaceRx } from '@cennznet/api/types';
+import { DeriveTokenInfo } from '@cennznet/api/derives/nft/types';
+import { Listing, TokenId } from '@cennznet/types';
+import { Option } from '@polkadot/types/codec/Option';
+
+/**
+ * Gets all tokens in a collection that have an open listing
+ *
+ * @param collectionId  The collection Id value
+ *
+ * @returns List of tokens that have open listings in given collection
+ */
+export function openCollectionListings(instanceId: string, api: ApiInterfaceRx) {
+  return (collectionId: string): Observable<DeriveTokenInfo[]> => {
+    return api.query.nft.openCollectionListings.keys(collectionId).pipe(
+      switchMap(
+        (storageKeys): Observable<any> => {
+          const listingIDs = storageKeys.map((storageKey) => {
+            return storageKey.args.map((k) => k.toHuman())[1];
+          });
+          return api.query.nft.listings.multi(listingIDs.map((listingID) => listingID)).pipe(
+            switchMap(
+              (listingsCodec: Option<Listing>[]): Observable<any> => {
+                const tokenIds = listingsCodec.map((listingCodec) => {
+                  const listing: Listing = listingCodec.unwrapOrDefault();
+                  const tokenID: TokenId = listing.asAuction.toJSON().tokens[0];
```

Comment:
listings can happen for `fixedPriceSale`s also

Reply:
ahh right cool, i'll add that as well